op-node/rollup/derive/span_channel_out.go

```diff
@@ -76,6 +105,9 @@ func (co *SpanChannelOut) Reset() error {
 	co.rlp[1].Reset()
 	co.lastCompressedRLPSize = 0
 	co.compressed.Reset()
+	if co.compressionAlgo == BROTLI {
+		co.compressed.WriteByte(0b0001)
+	}
```

Comment:
> And we probably only want to call this in co.compress() and not in Reset.

What's the reason for not adding the channel version when Reset()?

Reply:
nvm, that'll lead to the version byte being written twice

Reply:
`WritePrefix` I'm not sure if this works actually, because the channel version isn't written with the compressor, it should be written into the buffer already before the compression content.

Reply:
I think it's only fully abstracted if moving `compressed` to the compressor as well. Trying it out